supernode/store/store_mgr.go

```diff
@@ -38,21 +40,52 @@ func Register(name string, builder StorageBuilder) {
 
 // Manager manage stores.
 type Manager struct {
+	cfg *config.Config
+
+	defaultStorage *Store
+	mutex          sync.Mutex
 }
 
 // NewManager create a store manager.
-func NewManager() (*Manager, error) {
-	return &Manager{}, nil
+func NewManager(cfg *config.Config) (*Manager, error) {
+	return &Manager{
+		cfg: cfg,
+	}, nil
 }
 
 // Get a store from manager with specified name.
 func (sm *Manager) Get(name string) (*Store, error) {
 	v := plugins.GetPlugin(config.StoragePlugin, name)
 	if v == nil {
+		if name == LocalStorageDriver {
+			return sm.getDefaultStorage()
+		}
 		return nil, fmt.Errorf("not existed storage: %s", name)
 	}
 	if store, ok := v.(*Store); ok {
 		return store, nil
 	}
 	return nil, fmt.Errorf("get store error: unknown reason")
 }
+
+func (sm *Manager) getDefaultStorage() (*Store, error) {
+	if sm.defaultStorage != nil {
+		return sm.defaultStorage, nil
+	}
+
+	sm.mutex.Lock()
+	defer sm.mutex.Unlock()
+
+	if sm.defaultStorage == nil {
```

Comment:
```
if sm.defaultStorage != nil {
    return sm.defaultStorage, nil
} 
if sm.cfg == nil {
    return nil, fmt.Errorf("cannot init local storage without home path")
}
......
return sm.defaultStorage, nil
```
How about using this to reduce ident?

Reply:
actually, the initial version is as same as you wrote above...